### ISSUE TITLE
ErrorResponseFilter signature not consistent with global request/response filters

### ISSUE THREAD

**Issue by ludydoo** [ServiceStack/Issues#669]
Global request/response filters have a signature of `(request, response, responseDto)=>object`

`GetPlugin<ValidationFeature>().ErrorResponseFilter` has a signature of `(result, object) => object`

It would be helpful to have the `request` / `response` passed into the signature of the `Validation.ErrorResponseFilter`.

My usecase is :

Add a `globalRequestFilter` and start a `Stopwatch` , add it to the `request.items` under the `"stopWatch"` key.

Add a `globalResponseFilter`, check if the response dto has the "ResponseTime" property, set this property to ` Stopwatch.getElapsedMilliseconds`

The problem is when validation fails under the ValidationFeature, we can observe the results with the `Validation.ErrorResponseFilter`, but contrary to other filters, there's no way of knowing for which request the validation has failed.

Looking at the code (ValidationFilters.cs), adding these parameters to the Validation.ErrorResponseFilter would be trivial as they are readily available in the method triggering the ErrorResponseFilter :

```
private static async Task RequestFilterAsync(IRequest req, IResponse res, object requestDto,
            bool treatInfoAndWarningsAsErrors)

                    ...

                    if (validationFeature?.ErrorResponseFilter != null)
                    {
                        errorResponse = validationFeature.ErrorResponseFilter(validationResult, errorResponse);
                        // Could be changed to 
                        errorResponse = validationFeature.ErrorResponseFilter(req, res, validationResult, errorResponse);
                    }
```

**Comment by mythz** [comment]
I've added the `IRequest` in [this commit](https://github.com/ServiceStack/ServiceStack/commit/32d273bddd2cd4f9459ab77362aa6207029cde44) it doesn't need the `IResponse` as you can get it from `req.Response`.

This change is available from v5.5.1 that's now [available on MyGet](https://docs.servicestack.net/myget).

**Comment by ludydoo** [comment]
Thanks!
